src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java

```diff
@@ -79,14 +79,197 @@
+ * </p>
+ * <pre>
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Ignored due to default excludedClasses property
+ *   Map map = new HashMap(); // Ignored due to default excludedClasses property
+ *   Date date = new Date(); // Counted, 1
+ *   Time time = new Time(); // Counted, 2
+ *   // mention of 18 other user defined classes
+ *   // Counted, 20
+ *   Place place = new Place(); // Violation
+ * }
+ * </pre>
+ * <p>
+ * To configure the check with a threshold of 2:
+ * </p>
+ * <pre>
+ * &lt;module name="ClassFanOutComplexity"&gt;
+ *   &lt;property name="max" value="2"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code examples:
+ * </p>
+ * <p>
+ * The check passes without violations in the following.
+ * </p>
+ * <pre>
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Ignored due to default excludedClasses property
+ *   Map map = new HashMap(); // Ignored due to default excludedClasses property
+ *   Date date = new Date(); // Counted, 1
+ *   Time time = new Time(); // Counted, 2
+ * }
+ * </pre>
+ * <p>
+ * The check results in a violation in the following.
+ * </p>
+ * <pre>
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Ignored due to default excludedClasses property
+ *   Map map = new HashMap(); // Ignored due to default excludedClasses property
+ *   Date date = new Date(); // Counted, 1
+ *   Time time = new Time(); // Counted, 2
+ *   Place place = new Place(); // Violation
+ * }
+ * </pre>
+ * <p>
+ * To configure the check with three excluded classes {@code HashMap},
+ * {@code HashSet} and {@code Place}:
  * </p>
  * <pre>
  * &lt;module name="ClassFanOutComplexity"&gt;
- *   &lt;property name="max" value="10"/&gt;
+ *   &lt;property name="excludedClasses" value="HashMap, HashSet, Place"/&gt;
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Code examples:
+ * </p>
+ * <p>
+ * The check passes without violations in the following.
+ * </p>
+ * <pre>
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Set counted 1, HashSet ignored
+ *   Map map = new HashMap(); // Map counted 2, HashMap ignored
+ *   Date date = new Date(); // Counted, 3
+ *   Time time = new Time(); // Counted, 4
+ *   // mention of 16 other user defined classes
+ *   // Counted, 20
+ *   Place place = new Place(); // Ignored
+ * }
+ * </pre>
+ * <p>
+ * The check results in a violation in the following.
+ * </p>
+ * <pre>
+ * class InputClassComplexity {
+ *   Set set = new HashSet(); // Set counted 1, HashSet ignored
+ *   Map map = new HashMap(); // Map counted 2, HashMap ignored
+ *   Date date = new Date(); // Counted, 3
+ *   Time time = new Time(); // Counted, 4
+ *   // mention of 16 other user defined classes
+ *   // Counted, 20
+ *   Space space = new Space(); // Violation
+ * }
+ * </pre>
+ * <p>
+ * The following example demonstrates usage of <b>excludedClassesRegexps</b> property
+ * </p>
+ * <p>
+ * To configure the check with a regular expression {@code .*Reader$}:
```

Comment:
`To configure the check with a regular expression {@code .*Reader$}:`
-->
`To configure the check to exclude classes with a regular expression {@code .*Reader$}:`

Reply:
Done.

Reply:
Done.